tasks/task_manager.hh

```diff
@@ -126,7 +128,24 @@ public:
             }
 
             virtual future<> abort() noexcept {
-                return make_ready_future<>();
+                if (!_as.abort_requested()) {
+                    _as.request_abort();
+
+                    std::vector<task_info> children_info{_children.size()};
+                    boost::transform(_children, children_info.begin(), [] (const auto& child) {
+                        return task_info{child->id(), child.get_owner_shard()};
+                    });
+
+                    co_await coroutine::parallel_for_each(children_info, [this] (auto info) {
+                        return smp::submit_to(info.shard, [info, &tm = _module->get_task_manager().container()] {
+                            auto& tasks = tm.local().get_all_tasks();
+                            if (auto it = tasks.find(info.id); it != tasks.end()) {
+                                return it->second->abort();
+                            }
+                            return make_ready_future<>();
+                        });
+                    });
+                }
```

Comment:
Please follow up by moving all of impl (and anything else you can) to .cc. See #1.

Reply:
Done here: #12064.